sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/LeaseAsyncApiTests.java

```diff
@@ -0,0 +1,850 @@
+    public void breakBlobLeaseAC(OffsetDateTime modified, OffsetDateTime unmodified, String match, String noneMatch,
+                                 String tags) {
+        BlobAsyncClientBase bc = createBlobAsyncClient();
+        Map<String, String> t = new HashMap<>();
+        t.put("foo", "bar");
+        bc.setTags(t).block();
+        match = setupBlobMatchCondition(bc, match);
+        setupBlobLeaseCondition(bc, RECEIVED_ETAG);
+        BlobLeaseRequestConditions mac = new BlobLeaseRequestConditions()
+            .setIfModifiedSince(modified)
+            .setIfUnmodifiedSince(unmodified)
+            .setIfMatch(match)
+            .setIfNoneMatch(noneMatch)
+            .setTagsConditions(tags);
+
+        assertAsyncResponseStatusCode(createLeaseAsyncClient(bc).breakLeaseWithResponse(
+            new BlobBreakLeaseOptions().setRequestConditions(mac)), 202);
+    }
+
+    @ParameterizedTest
+    @MethodSource("acquireBlobLeaseACFailSupplier")
+    public void breakBlobLeaseACFail(OffsetDateTime modified, OffsetDateTime unmodified, String match, String noneMatch,
+                                     String tags) {
+        BlobAsyncClientBase bc = createBlobAsyncClient();
+        noneMatch = setupBlobMatchCondition(bc, noneMatch);
+        setupBlobLeaseCondition(bc, RECEIVED_LEASE_ID);
+        BlobLeaseRequestConditions mac = new BlobLeaseRequestConditions()
+            .setIfModifiedSince(modified)
+            .setIfUnmodifiedSince(unmodified)
+            .setIfMatch(match)
+            .setIfNoneMatch(noneMatch)
+            .setTagsConditions(tags);
+
+        StepVerifier.create(createLeaseAsyncClient(bc).breakLeaseWithResponse(new BlobBreakLeaseOptions().
+            setRequestConditions(mac)))
+            .verifyError(BlobStorageException.class);
+    }
+
+    @Test
+    public void breakBlobLeaseError() {
+        BlobAsyncClientBase bc = createBlobAsyncClient();
+        StepVerifier.create(createLeaseAsyncClient(bc).breakLease())
+            .verifyError(BlobStorageException.class);
+    }
+
+    @Test
+    public void changeBlobLease() {
+        BlobAsyncClientBase bc = createBlobAsyncClient();
+        BlobLeaseAsyncClient leaseClient = createLeaseAsyncClient(bc, testResourceNamer.randomUuid());
+        leaseClient.acquireLease(15).block();
+
+        String newLeaseId = testResourceNamer.randomUuid();
+        Response<String> changeLeaseResponse = leaseClient.changeLeaseWithResponse(
+            new BlobChangeLeaseOptions(newLeaseId)).block();
+
+        assertEquals(changeLeaseResponse.getValue(), newLeaseId);
+        assertEquals(changeLeaseResponse.getValue(), leaseClient.getLeaseId());
+
+        BlobLeaseAsyncClient leaseClient2 = createLeaseAsyncClient(bc, changeLeaseResponse.getValue());
+
+        assertAsyncResponseStatusCode(leaseClient2.releaseLeaseWithResponse(new BlobReleaseLeaseOptions()),
+            200);
+        assertTrue(validateBasicHeaders(changeLeaseResponse.getHeaders()));
```

Comment:
We should think about changing how `validateBasicHeaders` works to not return a boolean but to perform a set of asserts as that will more clearly tell us what failed rather than telling us we got false.

This is another change we can make in a follow-up PR.